### PR TITLE
`random` doc fix

### PR DESCRIPTION
**Pull request by albanD**
Fix the doc for random because it is actually inclusive.
Also removes some `torch.` functions that are not implemented and would always raise an error (they are not mentioned in the doc for `torch.`)

### CODE REVIEW

**Comment by apaszke** [comment]
Isn't random always exclusive? I thought I fixed that recently

**Comment by albanD** [comment]
Right now, it is actually inclusive. I added a test to check that (I think the probability of this test failing because of bad luck is low enough).

**Comment by apaszke** [comment]
Ugh no, it seems to be a mess. `.random_(3)` returns values in range [0, 2], but `.random_(1, 3)` uses [1, 3]... Can you just remove the +1 in [this line](https://github.com/pytorch/pytorch/blob/master/torch/csrc/generic/methods/TensorRandom.cwrap#L19).

**Comment by apaszke** [comment]
But that test was useful! Only adapt it to check exclusive ranges and check all possible dispatch options.

**Comment by albanD** [comment]
Ho, sorry, I thought the `Ugh no, it seems to be a mess.` was refering to the test!, I'll improve it and add it back.

File: test/test_torch.py [hunk 1]
```diff
@@ -840,6 +840,22 @@ def test_randperm(self):
         torch.randperm(100, out=res2)
         self.assertEqual(res1, res2, 0)
 
+    def test_random(self):
+        # This test is flaky with p<=(2/(ub-lb))^200=6e-36
+        t = torch.FloatTensor(200)
+        lb = 0
```
Comment (apaszke) [inline test/test_torch.py]: can you please use sth else than the default for `lb`? 1 would be better

**Review by soumith — APPROVED** [review]

**Comment by soumith** [comment]
thanks!

**Pull request merged**